Source/Ensure.Net/Ensure.cs

```diff
@@ -42,29 +43,39 @@ public static class Ensure
         /// Determine whether an object is null, if so throws a ArgumentNullException
         /// </summary>
         /// <param name="value">The object to be checked.</param>
-        public static void NotNull(object value)
+        public static void NotNull(object value, string parameterName)
         {
+            if (string.IsNullOrEmpty(parameterName))
+            {
+                throw new ArgumentException("Variable name cannot be null.");
+            }
+
             if (value == null)
             {
-                throw new ArgumentNullException("Variable cannot be null.");
+                throw new ArgumentNullException("", "Variable cannot be null.");
             }
         }
         /// <summary>
         /// Determine whether a string is null, if so throws an ArgumentNullException
         /// Determine whether a string is empty, if so throws an ArgumentException
         /// </summary>
         /// <param name="value">The string variable to be checked.</param>
-        public static void NotNullOrEmpty(string value)
+        public static void NotNullOrEmpty(string value, string parameterName)
         {
+            if (string.IsNullOrEmpty(parameterName))
```

Comment:
We should do either this check or the two checks bellow.  I think the checks below are the ones to keep.